AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/KMSKeystoreOperations.dfy

```diff
@@ -96,6 +96,51 @@ module {:options "/functionSyntax:4" } KMSKeystoreOperations {
     case kmsMRKeyArn(arn) => arn
   }
 
+  method GenerateRandom(
+    kmsClient: KMS.IKMSClient
+  )
+    returns (res: Result<KMS.GenerateRandomResponse, KmsError>)
+    requires kmsClient.ValidState()
+    modifies kmsClient.Modifies
+    ensures kmsClient.ValidState()
+    ensures
+      && |kmsClient.History.GenerateRandom| == |old(kmsClient.History.GenerateRandom)| + 1
+      && KMS.GenerateRandomRequest(
+           NumberOfBytes:=  Some(32),
+           CustomKeyStoreId:= None,
+           Recipient:= None
+         )
+         == Seq.Last(kmsClient.History.GenerateRandom).input
+      && old(kmsClient.History.GenerateRandom) < kmsClient.History.GenerateRandom
+      && old(kmsClient.History.Encrypt) == kmsClient.History.Encrypt
+      && old(kmsClient.History.Decrypt) == kmsClient.History.Decrypt
+
+    ensures res.Success? ==>
+              && res.value.Plaintext.Some?
+              && |res.value.Plaintext.value| == 32
```

Comment:
If we refactor to only return the `seq<bytes>` (or what ever the Dafny syntax is),
we should ensure that returned sequence size is 32.